examples/dynamic_watcher.rs

```diff
@@ -14,6 +14,12 @@ async fn main() -> anyhow::Result<()> {
     tracing_subscriber::fmt::init();
     let client = Client::try_default().await?;
 
+    // If set will receive only the metadata for watched resources
+    let should_watch_meta = {
+        let v = env::var("WATCH_METADATA").unwrap_or_else(|_| "false".into());
+        v.parse::<bool>().ok().unwrap_or_else(|| false)
+    };
```

Comment:
```suggestion
    let watch_metadata = std::env::var("WATCH_METADATA").map(|s| s == "1").unwrap_or(false);
```
small simplification; less example code that's not super relevant to a user, but slightly stricter on usage

Reply:
hm, this name clashes with the exported watcher variant and it makes me think:
maybe it's better to call the exported fn `watcher_metadata` to signify the similarity. `watch_object` has specifically different semantics.

Reply:
IMO this makes sense to take via Clap, like we do for some other examples?

Reply:
happy to do either :) I don't have a strong opinion here so I'll go with what you both think is more representative.

re: `watch_object`, wasn't my first choice tbh, I had initially picked `metadata_watcher`  but bumped into a `module_name_repetition` lint. I've overridden it just for this fn and re-exported it from the runtime crate so it functions similarly to `watcher`.

Reply:
Personally I'd be fine either silencing `module_name_repetition` for that fn or just calling it `metadata`.